src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,79 @@
 package core.basesyntax;
 
+import java.util.Objects;
+
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    static final int INITIAL_CAPACITY = 16;
+    static final float LOAD_FACTOR = 0.75f;
+    private Node<K, V>[] table;
+    private int size;
+
+    public MyHashMap() {
+        table = new Node[INITIAL_CAPACITY];
+    }
 
     @Override
     public void put(K key, V value) {
-
+        if (size >= table.length * LOAD_FACTOR) {
+            resize();
+        }
+        Node node = new Node(key, value, null);
+        int index = (key == null) ? 0 : Math.abs(key.hashCode()) % table.length;
```

Comment:
You can create a method where you create hashCode because you use this logic twice. At the beginning of this method and in the beginning of getValue()